src/RPC.ts

```diff
@@ -64,9 +64,7 @@ export default abstract class RPC {
     return new Promise((resolve, reject) => {
       const resolver: Resolver = (err, res) => {
         if (err) {
-          console.error(err);
-
-          reject(new Error(err.message));
+          reject(err);
```

Comment:
Did you change the way we were throing errors on the provider? Cause we relied on the error this

Reply:
Just checked this and reversed the change to avoid any issues 🚀